gs-rest-client/src/main/java/client/SimpleClient.java

```diff
@@ -118,95 +118,132 @@ public static void fixMyBug(ServerRequest serverRequest, TokenizerBuilder bobThe
+    * @param method: the method requested from the server
+    * 
+    * @return void: Simply print returned data so that the user can see it.
+    */
+    public static void handleResponse(String json, TokenizerBuilder tokenBuilder, String method) {
+        //Setup a new object mapper for converting json to POJO
+        ObjectMapper mapper = new ObjectMapper();
+
+        //Convert returned JSON string to a list of DatabaseEntry objects.
+        try {
+            DatabaseEntryListWrapper dbEntries = mapper.readValue(json, DatabaseEntryListWrapper.class);
 
-                System.out.println(bugFixes.getEntryList());
+            //Handle returned dbEntries based on requested server method.
+            switch (method) {
+            case "fix":
+                System.out.println(dbEntries.getEntryList());
 
-                for (DatabaseEntry e : bugFixes.getEntryList()) {
+                for (DatabaseEntry e : dbEntries.getEntryList()) {
                     System.out.println("\nFixed Code:");
-                    System.out.println(bobTheBuilder.harmonize(e.getFixedCode()));
+                    System.out.println(tokenBuilder.harmonize(e.getFixedCode()));
                 }
-                conn.disconnect();
+                break;
+            case "echo":
+                System.out.println(dbEntries.getEntryList());
+                break;
+            case "index":
+                System.out.println(json.toString());
+                break;
+            default:
+                System.out.println(dbEntries.getEntryList());
+
+                for (DatabaseEntry e : dbEntries.getEntryList()) {
+                    System.out.println("\nFixed Code:");
+                    System.out.println(tokenBuilder.harmonize(e.getFixedCode()));
+                }
+                break;
             }
 
-        } catch (MalformedURLException e) {
-            e.printStackTrace();
         } catch (IOException e) {
             e.printStackTrace();
         }
     }
 
     public static void main(String[] args) {
-
-        if(args.length != 4) {
-            System.out.println("Usage: java -jar <jar> <file> <line # start> <line # end> <server" +
-                    " method> (fix, test, index)");
+        //Grab arguments from the command line and setup variables
+        if(args.length != 5) {
+            System.out.println("Usage: java -jar <jar> <file> <error> <line # start> <line # end> <server" +
+                    " method> (fix, echo, index)");
             System.exit(0);
         }
 
         String fileName = args[0];
-        int startLine = Integer.parseInt(args[1]);
-        int endLine = Integer.parseInt(args[2]);
-        String method = args[3];
-
+        String errorMessage = args[1];
+        int startLine = Integer.parseInt(args[2]);
+        int endLine = Integer.parseInt(args[3]);
+        String method = args[4];
         ServerRequest serverRequest;
 
-        if (method.equals("index")) {
```

Comment:
The problem is that this is going to kick and crash if filename doesn't resolve to something on your system, ultimately this is a problem with the way we are handling arguments and should probably be treated by rewriting this to be robust and not demand 5 arguments for specific things